taichi/program/state_flow_graph.cpp

```diff
@@ -790,14 +801,18 @@ std::unordered_set<int> StateFlowGraph::fuse_range(int begin, int end) {
     if (!fused[i]) {
       // Fuse no more than one task into task i
       bool i_updated = false;
+      int remaining_fuses = get_initial_remaining_fuses();
       for (auto &edge : nodes[i]->output_edges.get_all_edges()) {
         if (edge.second->pending()) {
           const int j = edge.second->pending_node_id - begin;
           if (j >= 0 && j < nodes.size() && edge_fusible(i, j)) {
             do_fuse(i, j);
             // Iterators of nodes[i]->output_edges may be invalidated
             i_updated = true;
```

Comment:
IIUC, `i_updated` isn't useful, and we can remove L819-821?

Reply:
Oh yes. I thought there were nested for-loops when I wrote this but there weren't.